catalyst-gateway/bin/src/utils/blake2b_hash.rs

```diff
@@ -0,0 +1,134 @@
+//! Types of Blake-2b Hash
+
+use blake2b_simd::Params;
+
+/// Generates a UUID string from the provided key and data using `BLAKE2b` hashing.
+///
+/// # Arguments
+/// - `key`: A string slice that is used as part of the hash function input.
+/// - `data`: A vector of strings which will be included in the `BLAKE2b` hash
+///   computation.
+///
+/// # Returns
+/// A UUID string generated from the `BLAKE2b` hash of the concatenated data with the key.
+pub(crate) fn generate_uuid_string_from_data(key: &str, data: Vec<String>) -> String {
```

Comment:
I think we can change `data: Vec<String>` to `data: &[String]`. It has nothing to do with consuming a `Vec<T>`, the vector itself is purely readonly.